### PR TITLE
chore: Update to latest snapd client

### PR DESCRIPTION
**Pull request by spydon**
This is still using a git dependency for snapd.dart, but it will be published as soon as it is verified that all the types work.

### CODE REVIEW

File: packages/app_center/test/app_card_test.dart [hunk 1]
```diff
@@ -14,10 +13,17 @@ const snapRating = Rating(
...
   name: 'testsnap',
   id: 'r4LxMVp7zWramXsJQAKdamxy6TAWlaDD',
   summary: 'Its a summary!',
+  revision: '',
+  version: '',
+  channel: '',
+  base: '',
+  contact: '',
+  description: '',
+  type: '',
```
Comment (d-loose) [inline packages/app_center/test/app_card_test.dart:26]: Those aren't needed here as they correspond to the defaults in `createSnap`

**Comment by d-loose** [comment]
Works fine :+1:
Just wondering if it's a good idea to mark some additional fields in `Snap` as required (e.g. `contact` and `description`). Maybe it's better to keep them optional and provide empty defaults? I'll have a closer look at the fields in the corresponding snapd.dart PR

File: packages/app_center/test/games_page_test.dart [hunk 2]
```diff
@@ -25,9 +25,9 @@ void main() {
     const SnapSearchParameters(query: 'testsn'): [
       createSnap(name: 'testsnap', title: 'Test Snap', downloadSize: 3),
       createSnap(
-          name: 'testsnap2', title: 'Another Test Snap', downloadSize: 1),
+          name: 'testsnap2', title: 'Another Test Snap', downloadSize: 1,),
```
Comment (d-loose) [inline packages/app_center/test/games_page_test.dart:28]: Could you run `melos format`?
Seems like the CI doesn't complain about this. I think we used to run `dart format --set-exit-if-changed` before switching to the melos action.

Comment (spydon) [reply packages/app_center/test/games_page_test.dart:28]: Whaaat, I'll fix that, not much sense in running formatting in the pipeline if it doesn't complain :smile:

**Review by d-loose — APPROVED** [review]

File: packages/app_center/test/games_page_test.dart (hunk 2 above)
Comment (spydon) [reply packages/app_center/test/games_page_test.dart]: Added to check in https://github.com/ubuntu/app-center/pull/1716

**Pull request merged**